### PR TITLE
Adds new projects

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 16
           

Complexity increasing per file
==============================
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/libssh.py  3
         

Complexity decreasing per file
==============================
+ varats/utils/experiment_util.py  -3
         

Clones added
============
- varats/projects/c_projects/bison.py  1
- varats/projects/c_projects/gawk.py  4
- varats/projects/c_projects/qemu.py  2
- varats/projects/c_projects/opus.py  1
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/libpng.py  3
- varats/experiments/vara/vara_full_mtfa.py  1
- varats/projects/test_projects/taint_tests.py  1
- varats/projects/c_projects/openvpn.py  3
- varats/projects/test_projects/basic_tests.py  1
- varats/experiments/vara/vara_fc_taint_analysis.py  1
- varats/projects/c_projects/libssh.py  3
         

Clones removed
==============
+ varats/projects/c_projects/busybox.py  -7
+ varats/projects/c_projects/redis.py  -8
+ varats/projects/c_projects/htop.py  -6
+ varats/projects/c_projects/openssl.py  -3
+ varats/projects/c_projects/gzip.py  -3
+ varats/projects/c_projects/tmux.py  -2
+ varats/projects/c_projects/curl.py  -3
+ varats/projects/c_projects/lz4.py  -7
+ varats/projects/c_projects/vim.py  -3
+ varats/projects/c_projects/glibc.py  -4
+ varats/projects/c_projects/x264.py  -2
+ varats/projects/c_projects/coreutils.py  -3
+ varats/projects/c_projects/libvpx.py  -2
+ varats/projects/c_projects/xz.py  -4
+ varats/projects/c_projects/gravity.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 17
           

Complexity increasing per file
==============================
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/irssi.py  3
         

Complexity decreasing per file
==============================
+ varats/utils/experiment_util.py  -3
         

Clones added
============
- varats/experiments/vara/vara_full_mtfa.py  1
- varats/projects/c_projects/qemu.py  2
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/bitlbee.py  6
- varats/experiments/vara/vara_fc_taint_analysis.py  1
- varats/projects/c_projects/bison.py  1
- varats/projects/c_projects/gawk.py  4
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/test_projects/taint_tests.py  1
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/opus.py  1
- varats/projects/test_projects/basic_tests.py  1
         

Clones removed
==============
+ varats/projects/c_projects/tmux.py  -2
+ varats/projects/c_projects/x264.py  -2
+ varats/projects/c_projects/glibc.py  -4
+ varats/projects/c_projects/gzip.py  -3
+ varats/projects/c_projects/coreutils.py  -3
+ varats/projects/c_projects/libvpx.py  -2
+ varats/projects/c_projects/openssl.py  -3
+ varats/projects/c_projects/busybox.py  -7
+ varats/projects/c_projects/gravity.py  -3
+ varats/projects/c_projects/redis.py  -8
+ varats/projects/c_projects/xz.py  -4
+ varats/projects/c_projects/htop.py  -5
+ varats/projects/c_projects/vim.py  -3
+ varats/projects/c_projects/lz4.py  -7
+ varats/projects/c_projects/curl.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/openvpn.py  3
         

Clones added
============
- varats/projects/c_projects/bison.py  1
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/c_projects/gawk.py  4
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/c_projects/openvpn.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/bison.py  3
         

Clones added
============
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/c_projects/irssi.py  8
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/gawk.py  4
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/bison.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           

Complexity increasing per file
==============================
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/libxml2.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/libtiff.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/gawk.py  3
         

Clones added
============
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/libxml2.py  2
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/bison.py  1
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/c_projects/libtiff.py  4
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/gawk.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

File: varats/projects/cpp_projects/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            limit=None,
+            shallow=False,
+            version_filter=project_filter_generator("mongodb")
+        )
+    ]
+
+    @property
+    def binaries(self) -> tp.List[ProjectBinaryWrapper]:
+        """Return a list of binaries generated by the project."""
+        # TODO: please add correct binary names
```
Comment (vulder) [inline varats/projects/cpp_projects/mongodb.py:39]: fix me

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libxml2.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/libtiff.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/cpp_projects/mongodb.py  3
         

Clones added
============
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/gawk.py  1
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libxml2.py  2
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/libtiff.py  4
- varats/projects/c_projects/libssh.py  3
- varats/projects/cpp_projects/mongodb.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Review by boehmseb — APPROVED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/libtiff.py  3
- varats/projects/c_projects/libxml2.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libpng.py  3
         

Clones added
============
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/libtiff.py  4
- varats/projects/c_projects/libxml2.py  2
- varats/projects/c_projects/gawk.py  1
- varats/projects/c_projects/openvpn.py  3
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libpng.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/libxml2.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/bitlbee.py  3
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libtiff.py  3
         

Clones added
============
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/libxml2.py  2
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/gawk.py  1
- varats/projects/c_projects/bitlbee.py  6
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libtiff.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/projects/cpp_projects/mongodb.py  3
- varats/projects/c_projects/libtiff.py  3
- varats/projects/c_projects/libxml2.py  3
- varats/projects/c_projects/irssi.py  3
- varats/projects/c_projects/gawk.py  3
- varats/projects/c_projects/bison.py  3
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/bitlbee.py  3
         

Clones added
============
- varats/projects/cpp_projects/mongodb.py  1
- varats/projects/c_projects/libtiff.py  4
- varats/projects/c_projects/libxml2.py  2
- varats/projects/c_projects/irssi.py  8
- varats/projects/c_projects/gawk.py  1
- varats/projects/cpp_projects/poppler.py  3
- varats/projects/c_projects/libssh.py  3
- varats/projects/c_projects/openvpn.py  3
- varats/projects/c_projects/libpng.py  3
- varats/projects/c_projects/bitlbee.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5987952&bid=19636701)

**Pull request merged**